scripts/install_xcpng.py

```diff
@@ -232,9 +282,16 @@ def main():
         check_mac_address(host2, mac_address)
         logging.info('New host is started and enabled')
         if host2.xcp_version != version.parse(xcp_version):
+            if host2.xcp_version < version.parse(xcp_version) and args.action == 'restore':
+                logging.info(f'The host {vm_ip_address} is in version : {host2.xcp_version}')
+            else:
+                raise Exception(
+                    f"The installed host ({vm_ip_address}) is not in the expected version. Got: {host2.xcp_version}.\n"
+                    f"Expected: {xcp_version}."
+                )
+        elif host2.xcp_version == version.parse(xcp_version) and args.action == 'restore':
             raise Exception(
-                f"The installed host ({vm_ip_address}) is not in the expected version. Got: {host2.xcp_version}.\n"
-                f"Expected: {xcp_version}."""
+                f"The installed host ({vm_ip_address}) is not in a previous version. Got: {host2.xcp_version}.\n"
```

Comment:
It would be better to first test the action and then test the version accordingly:

```
if args.action == 'restore':
    # check that the new version is lower than xcp_version
else
    # check that the version is exactly the expected version
```

Reply:
Plus instead of imbricating `if` blocks. an assert would do the job here:
```py
if args.action == 'restore':
    assert(new_version < current_version, <error msg to log>)
else
    assert(new_version == current_version, <other error msg to log>)

Reply:
We're not in the code of a test so semantically speaking I wouldn't use assert to check the result of the operation, even if the result would be globally the same.

Reply:
Well, an assertion is also used to be sure an algorithm or function is correctly implemented. It's not reserved for testing. ;)

In the present case, I suppose we should use an exception if we can trigger this problem using a "bad" XCP-ng.

Reply:
Yes, the error would not be related to the implementation of the script, so Exception it is.